torch/_decomp/decompositions.py

```diff
@@ -3278,63 +3299,92 @@ def upsample_bilinear2d(
     scales_h: Optional[float] = None,
     scales_w: Optional[float] = None,
 ) -> Tensor:
-    # get dimensions of original image
-    n_batch, n_channels, in_h, in_w = input.shape
-
-    out_h = output_size[0]
-    out_w = output_size[1]
-
-    # Calculate horizontal and vertical scaling factor
-    # TODO: Figure out if scales_h/scales_w matters here
-    if out_h > 1:
-        if align_corners:
-            h_scale_factor = (in_h - 1) / (out_h - 1)
-        else:
-            h_scale_factor = 1.0 / scales_h if scales_h is not None else in_h / out_h
-    else:
-        h_scale_factor = 0.0
+    return _upsample_linear(input, output_size, align_corners, [scales_h, scales_w])
 
-    if out_w > 1:
-        if align_corners:
-            w_scale_factor = (in_w - 1) / (out_w - 1)
-        else:
-            w_scale_factor = 1.0 / scales_w if scales_w is not None else in_w / out_w
-    else:
-        w_scale_factor = 0.0
 
-    i = torch.arange(out_h, dtype=input.dtype, device=input.device)
-    j = torch.arange(out_w, dtype=input.dtype, device=input.device)
+@register_decomposition(
+    [aten.upsample_trilinear3d.default, aten.upsample_trilinear3d.out]
+)
+@aten.upsample_trilinear3d.default.py_impl(DispatchKey.Autograd)
```

Comment:
If you are implementing the backward in a different PR, you should remove this registration. Same for the other upsample ops.

Reply:
I guess that it'll be much slower than the previous implementation, so yeah, better to remove it if this is the case. If it's just about 50% slower it may be alright to leave it as fusions may very well make up for this.